where-to-travel/src/main/webapp/script.js

```diff
@@ -664,3 +659,204 @@ function addAcceptablePlaces(time, places, acceptablePlacesInfo) {
+ * information about place and populates this information inside place's infocard. 
+ *
+ * @param {String} place_id Textual identifier of place for PlaceDetails request
+ */
+function populateMorePlaceInfo(place_id) {
+  let request = {
+    placeId: place_id,
+    fields: [
+      'formatted_address',
+      'formatted_phone_number',
+      'geometry',
+      'opening_hours', 
+      'rating', 
+      'website' 
+    ]
+  };
+
+  placesService.getDetails(request, callback);
+
+  function callback(place, status) {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+      
+      // Left side of card contains (if available) rating, address,phone_number
+      const leftHTML = getLeftCardHTML(place);
+      // Right side of card contains listing of places' opening hours
+      const rightHTML = getRightCardHTML(place);
+      // Bottom of card contains button with link to directions, website (if available), and hide info.
+      const bottomHTML = getBottomCardHTML(place, place_id);
+
+      let newHTML = '' +
+        `<div class="container">
+           <div class="row">
+             <div class="col">
+               ${leftHTML}
+             </div>
+             <div class="col">
+              ${rightHTML}
+             </div>
+           </div>
+         </div>
+         ${bottomHTML}`;
+      
+      // Place content in infocard corresponding to place
+      document.getElementById(place_id).innerHTML = newHTML;
+    }
+  }
+}
+
+/**
+ * Places information about place including formatted address, and if available, rating and formatted
+ * phone number in HTML string that makes up content of left side of infocard corresponding to place.
+ * 
+ * @param {Object} place Contains (if available) rating, formatted address, and formatted phone number 
+ * @return {String} HTML content that formats passed in information for left side of infocard
+ */
+function getLeftCardHTML(place) {
+  let leftHTML = ``;
+
+  // If there is a rating, convert to percentage out of five and fill in stars according to percentage
+  if (place.rating) {
+    const percent = Math.round((place.rating/5) * 100);
+    leftHTML += `<div class="stars"><span style="width:${percent}%" class="stars-rating"></span></div><br/>`;
+  }
+
+  leftHTML += `<p><b>Address:</b> &nbsp ${place.formatted_address}</p>`;
+
+  if (place.formatted_phone_number) {
+    leftHTML += `<p><b>Phone:</b> &nbsp ${place.formatted_phone_number}</p>`;
+  }
+
+  return leftHTML;
+}
+
+/**
+ * Places information about opening hours, if available, in HTML string that makes up content of 
+ * right side of infocard corresponding to place.
+ * 
+ * @param {Object} place Contains (if available) opening hours of place
+ * @return {String} HTML content that formats passed in information for right side of infocard
+ */
+function getRightCardHTML(place) {
+  let rightHTML = ``;
+
+  if (place.opening_hours) {  
+    rightHTML = getOpeningHours(place.opening_hours);
+  }
+
+  return rightHTML;
```

Comment:
What do you think about replacing this with the following, instead of defining rightHtml?

`if (place.opening_hours) {  
    return getOpeningHours(place.opening_hours);
  } else {
    return '';
}`

LMK

Reply:
Not in one line of course, github bad at formatting

Reply:
I think that's a good idea! I have changed it to a ternary expression.